### PR TITLE
fixes TestKspNotifsBulk test

### PR DESCRIPTION
**Pull request by badari31**
- changed the heartbeat interval to 2 seconds
- made heartbeat interval non-configurable
- changed `listenToKsps` test method which used to listen for expirations

### CODE REVIEW

File: impl/relredis.go [hunk 1]
```diff
@@ -184,22 +187,18 @@ func (r *ReliableRedisStreamClient) Done() {
 	ctx := context.Background()
 
 	for streamName := range r.streamLocks {
-		if err := r.DoneDataStream(ctx, streamName); err != nil {
-			log.Println("error", err, " occured while marking stream", streamName, " as done; moving on to other streams ...")
-		}
+		r.DoneDataStream(ctx, streamName)
```
Comment (akat-sukii) [inline impl/relredis.go:190]: Shouldn't we propogate the error back up?

File: impl/helpers.go [hunk 2]
```diff
@@ -55,3 +47,10 @@ func (r *ReliableRedisStreamClient) lockAndExtend(mutex *redsync.Mutex) error {
 
 	return nil
 }
+
+func (r *ReliableRedisStreamClient) safeCloseLBS() {
+	if !r.lbsChanClosed {
```
Comment (akat-sukii) [inline impl/helpers.go:52]: This can still close race. This should be an atomic.

File: impl/relredis.go [hunk 3]
```diff
@@ -184,22 +187,18 @@ func (r *ReliableRedisStreamClient) Done() {
...
 
 	// release resources
 	r.cleanup()
 }
 
 func (r *ReliableRedisStreamClient) cleanup() {
-	close(r.lbsChan)
+	r.safeCloseLBS()
+
+	r.pubSub.Close()
```
Comment (akat-sukii) [inline impl/relredis.go:200]: Error should be propogated imo instead of silently ignoring.

File: .vscode/launch.json
Comment (akat-sukii) [inline .vscode/launch.json:1]: Let's not commit `.vscode`?

File: impl/helpers.go [hunk 4]
```diff
@@ -49,8 +49,8 @@ func (r *ReliableRedisStreamClient) lockAndExtend(mutex *redsync.Mutex) error {
 }
 
 func (r *ReliableRedisStreamClient) safeCloseLBS() {
-	if !r.lbsChanClosed {
+	if !r.lbsChanClosed.Load() {
 		close(r.lbsChan)
-		r.lbsChanClosed = true
+		r.lbsChanClosed.CompareAndSwap(false, true)
 	}
```
Comment (hthuwal) [inline impl/helpers.go:55]: The race still exists. Mutliple threads can execute line 52 and enter before 54 executes. It should be

```suggestion
	if r.lbsChanClosed.CompareAndSwap(false, true) {
		close(r.lbsChan)
	}
```

**Review by hthuwal — DISMISSED** [review]
LGTM!

**Review by hthuwal — APPROVED** [review]

**Pull request merged**